dao/courses.go

```diff
@@ -96,15 +96,35 @@ func GetPublicCourses(year int, term string) (courses []model.Course, err error)
 		return cachedCourses.([]model.Course), err
 	}
 	var publicCourses []model.Course
+
 	err = DB.Preload("Streams", func(db *gorm.DB) *gorm.DB {
 		return db.Order("start asc")
-	}).Find(&publicCourses, "visibility = 'public' AND teaching_term = ? AND year = ?", term, year).Error
+	}).Find(&publicCourses, "visibility = 'public' AND teaching_term = ? AND year = ?",
+		term, year).Error
+
 	if err == nil {
 		Cache.SetWithTTL(fmt.Sprintf("publicCourses%v%v", year, term), publicCourses, 1, time.Minute)
 	}
 	return publicCourses, err
 }
 
+func GetPublicAndLoggedInCourses(year int, term string) (courses []model.Course, err error) {
+	cachedCourses, found := Cache.Get(fmt.Sprintf("publicAndLoggedInCourses%v%v", year, term))
+	if found {
+		return cachedCourses.([]model.Course), err
+	}
+	var publicCourses []model.Course
+
+	err = DB.Preload("Streams", func(db *gorm.DB) *gorm.DB {
+		return db.Order("start asc")
+	}).Find(&publicCourses,
+		"(visibility = 'public' OR visibility = 'loggedin') AND teaching_term = ? AND year = ?", term, year).Error
+	if err == nil {
+		Cache.SetWithTTL(fmt.Sprintf("publicAndLoggedInCourses%v%v", year, term), publicCourses, 1, time.Minute)
```

Comment:
nit: I did this wrong as well and this works just fine, but the right format string would be `publicAndLoggedInCourses%d%s`